spec/service/cloud_watch_service_spec.rb

```diff
@@ -66,4 +67,68 @@
       end
     end
   end
+
+  describe "#week_starts" do
+    let(:datapoints) { [{ sum: total_starts }] }
+    let(:total_starts) { 5.0 }
+
+    around do |example|
+      Timecop.freeze(Time.zone.local(2021, 1, 1, 4, 30, 0)) do
+        example.run
+      end
+    end
+
+    before do
+      allow(Settings).to receive(:forms_env).and_return(forms_env)
+    end
+
+    it "calls the cloudwatch client with get_metric_statistics" do
+      cloudwatch_client = Aws::CloudWatch::Client.new(stub_responses: true)
+      metric_response = cloudwatch_client.stub_data(:get_metric_statistics, datapoints:)
+
+      allow(Aws::CloudWatch::Client).to receive(:new).and_return(cloudwatch_client)
+
+      allow(cloudwatch_client).to receive(:get_metric_statistics).with({
+        metric_name: "started",
+        namespace: "forms/#{forms_env}",
+        dimensions: [
+          {
+            name: "form_id",
+            value: form_id.to_s,
+          },
+        ],
+        start_time: Time.zone.now.midnight - 7.days,
+        end_time: Time.zone.now.midnight,
+        period: 604_800,
+        statistics: %w[Sum],
+        unit: "Count",
+      }).and_return(metric_response)
+
+      described_class.week_starts(form_id:)
```

Comment:
is there meant to be an expectation in this spec?

Reply:
I was confused by this - I originally had line 91 start with an `expect` instead of `allow`. Rubocop told me off, and it turns out that the tests will still fail if the `allow` isn't triggered. I'm not a fan...

Reply:
You know what, after reading the docs for the cop, it looks like it's best practice to have both an allow and an expect. It comes out a bit weird imo, but at least there's an expect to hold on to.

https://www.rubydoc.info/gems/rubocop-rspec/RuboCop/Cop/RSpec/StubbedMock

Reply:
I've reworked it to use an expect as well as the allow, and I've also deduped some test setup bits